src/backend/mailbox_manager/models.py

```diff
@@ -2,17 +2,35 @@
 Declare and configure the models for the People additional application : mailbox_manager
 """
 
+from logging import getLogger
+
 from django.conf import settings
-from django.core import validators
-from django.core.exceptions import ValidationError
-from django.db import models
+from django.core import exceptions, validators
+from django.db import models, transaction
 from django.utils.text import slugify
 from django.utils.translation import gettext_lazy as _
 
+import requests
+from urllib3.util import Retry
+
 from core.models import BaseModel
 
 from mailbox_manager.enums import MailDomainRoleChoices, MailDomainStatusChoices
 
+logger = getLogger(__name__)
+
+adapter = requests.adapters.HTTPAdapter(
+    max_retries=Retry(
+        total=4,
+        backoff_factor=0.1,
+        status_forcelist=[500, 502],
+        allowed_methods=["PATCH"],
```

Comment:
Why PATCH? Because of idempotency? Shouldn't we retry on a "GET" too? 

Also, why would you retry on a 500?

Reply:
It's now in the `utils` folder but the question remains whole.